rosdoc2/verbs/build/parse_rosdoc2_yaml.py

```diff
@@ -57,6 +57,17 @@ def parse_rosdoc2_yaml(yaml_string, build_context):
             f"expected a dict{{output_dir: build_settings, ...}}, "
             f"got a '{type(builders_dict)}' instead")
 
+    if 'python_source' in settings_dict:
+        build_context.add_python_source(python_source=settings_dict['python_source'])
+
+    if 'always_run_doxygen' in settings_dict:
+        build_context.add_always_run_doxygen(
+            always_run_doxygen=settings_dict['always_run_doxygen'])
+
+    if 'always_run_sphinx_apidoc' in settings_dict:
+        build_context.add_always_run_sphinx_apidoc(
+            always_run_sphinx_apidoc=settings_dict['always_run_sphinx_apidoc'])
+
```

Comment:
This pattern is weird to me. There are places where you unconditionally do things like `build_context.always_run_doxygen`, but the way this is set up if the matching case here were false, that code would result in an error because the build context will have no member `always_run_doxygen`.

Wouldn't it be better to give default values (maybe just `None`) to these members in the constructor of the build context and just set them here if they show up in the settings dict?

Maybe there's a reason you're doing it this way, but it seems like we're misusing the build context as a class.

Reply:
Hmm, my bad. I conflated the absence of a class member (i.e. a type does **not** have an attribute) with that attribute just having a `None` value. I will address that with default `None` initialization in the build_context constructor.

---

About the use of the build context in this manner: I am using it in this way to propagate information from the build invocation through to each of the builders (Doxygen, Sphinx). I wasn't sure of another way to exhibit this behavior, and I believe settings like `always_run_doxygen` do fall under the scope of build context. Having said that, we could very well just introduce a new object type to propagate this information throughout `rosdoc2`, or maybe just introduce/read these settings elsewhere. I'm not convinced they're better than the current use of the build context as a class though. Thoughts?

Reply:
I think it's fine to pass additional information via the build context in this way.